docs/guides/gitlab.md

```diff
@@ -0,0 +1,200 @@
+    ...
+    nginx['ssl_certificate'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io.pem"
+    nginx['ssl_certificate_key'] = "/etc/gitlab/ssl/gitlab.localhost.pomerium.io-key.pem"
+    ```
+
+1. Reconfigure GitLab to use the new configuration:
+
+    ```bash
+    docker exec -u 0 -it gitlab gitlab-ctl reconfigure
+    ```
+
+## Configure a Pomerium Route
+
+Edit `config.yaml` and add a route for GitLab. Note that this example assumes access to the mkcert CA certificate at `/mkcert/rootCA.pem`:
+
+```yaml
+  - from: https://gitlab.localhost.pomerium.io
+    to: https://gitlab-ee
+    pass_identity_headers: true
+    tls_custom_ca_file: /mkcert/rootCA.pem
+    tls_server_name: gitlab.localhost.pomerium.io
+    preserve_host_header: true
+    policy:
+      - allow:
+          or:
+            - domain:
+                is: example.com
+```
+
+Once the route is applied, you should be able to access GitLab from `https://gitlab.localhost.pomerium.io`:
+
+Use `grep` within the container to find the default root password:
+
+```bash
+sudo docker exec -it gitlab grep 'Password:' /etc/gitlab/initial_root_password
+```
+
+### Configure TCP Connections
+
+1. An additional route will provide an encrypted TCP tunnel through which users can securely access code using Git:
+
+  ```yaml
+    - from: tcp+https://gitlab.localhost.pomerium.io
+      to: tcp://gitlab-ee:22
+      pass_identity_headers: true
+      tls_custom_ca_file: /mkcert/rootCA.pem
+      tls_server_name: gitlab.localhost.pomerium.io
+      preserve_host_header: true
+      policy:
+        - allow:
+            or:
+              - groups:
+                  has: devs@example.com
+  ```
+
+1. Once this route is applied, users can create an encrypted connection using [pomerium-cli] or the [Pomerium Desktop] app:
+
+    ::::: tabs
+    :::: tab Pomerium-CLI
+    ```bash
+    pomerium-cli tcp gitlab.pomerium.localhost.io:22 --listen :2202
+    ```
+
+    The `--listen` key defines what port the tunnel listens at locally.
+    ::::
+    :::: tab Pomerium Desktop
+    ![An encrypted tunnel for git connections using Pomerium Desktop](./img/gitlab/gitlab-desktop-connection.png)
+    ::::
+    :::::
+
+1. Add the tunneled connection as a remote:
+
+    ```bash
+    git remote add gitlab-tunneled ssh://git@127.0.0.1:2202/username/project-name
+    ```
+
+Now when you first initiate a `pull`, `push`, or `fetch` command your web browser will open to authenticate and authorize the connection.
+
+## Identity Management
+
+While GitLab has a [JWT OmniAuth provider], it only accepts a json web token (**JWT**) as a callback url parameter and not as an HTTP header as [provided by pomerium](/reference/readme.md#pass-identity-headers). If your IdP is a [supported Omniauth provider](https://docs.gitlab.com/ee/integration/omniauth.html#supported-providers), you can integrate it directly to GitLab to re-use your current session, but it will require signing in twice, once with Pomerium and again with GitLab:
```

Comment:
```suggestion
While GitLab has a [JWT OmniAuth provider], it only accepts a JSON web token (**JWT**) as a callback url parameter and not as an HTTP header as [provided by pomerium](/reference/readme.md#pass-identity-headers). If your IdP is a [supported Omniauth provider](https://docs.gitlab.com/ee/integration/omniauth.html#supported-providers), you can integrate it directly to GitLab to re-use your current session, but it will require signing in twice, once with Pomerium and again with GitLab:
```